make/Images.gmk

```diff
@@ -85,29 +86,97 @@ JLINK_TOOL := $(JLINK) -J-Djlink.debug=true \
 
 JLINK_JRE_EXTRA_OPTS := --no-man-pages --no-header-files --strip-debug
 
+DEFAULT_JDK_IMAGE_SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/jdk
+ifeq ($(JLINK_PRODUCE_RUNTIME_LINK_JDK), true)
+  JDK_LINK_OUTPUT_DIR := $(SUPPORT_OUTPUTDIR)/images/runtime-link-initial-jdk
+  JDK_IMAGE_SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/runtime-link-support
+else
+  JDK_LINK_OUTPUT_DIR := $(JDK_IMAGE_DIR)
+  JDK_IMAGE_SUPPORT_DIR := $(DEFAULT_JDK_IMAGE_SUPPORT_DIR)
+endif
+
 ifeq ($(JLINK_KEEP_PACKAGED_MODULES), true)
-  JLINK_JDK_EXTRA_OPTS := --keep-packaged-modules $(JDK_IMAGE_DIR)/jmods
+  ifeq ($(JLINK_PRODUCE_RUNTIME_LINK_JDK), true)
+    JLINK_RUNTIME_CREATE_EXTRA := --keep-packaged-modules $(JDK_IMAGE_DIR)/jmods
+  endif
+  JLINK_JDK_EXTRA_OPTS := --keep-packaged-modules $(JDK_LINK_OUTPUT_DIR)/jmods
 endif
 
 JLINK_DISABLE_WARNINGS := | ( $(GREP) -v -e "WARNING: Using incubator module" || test "$$?" = "1" )
 
-JDK_IMAGE_SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/jdk
 JRE_IMAGE_SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/jre
 
 $(eval $(call SetupExecute, jlink_jdk, \
     WARN := Creating jdk image, \
     DEPS := $(JDK_JMODS) $(BASE_RELEASE_FILE) \
         $(call DependOnVariable, JDK_MODULES_LIST, $(JDK_IMAGE_SUPPORT_DIR)/_jlink_jdk.vardeps), \
-    OUTPUT_DIR := $(JDK_IMAGE_DIR), \
+    OUTPUT_DIR := $(JDK_LINK_OUTPUT_DIR), \
     SUPPORT_DIR := $(JDK_IMAGE_SUPPORT_DIR), \
-    PRE_COMMAND := $(RM) -r $(JDK_IMAGE_DIR), \
+    PRE_COMMAND := $(RM) -r $(JDK_LINK_OUTPUT_DIR), \
     COMMAND := $(JLINK_TOOL) --add-modules $(JDK_MODULES_LIST) \
-        $(JLINK_JDK_EXTRA_OPTS) --output $(JDK_IMAGE_DIR) \
+        $(JLINK_JDK_EXTRA_OPTS) --output $(JDK_LINK_OUTPUT_DIR) \
         $(JLINK_DISABLE_WARNINGS), \
 ))
 
 JLINK_JDK_TARGETS := $(jlink_jdk)
 
+ifeq ($(JLINK_PRODUCE_RUNTIME_LINK_JDK), true)
+
+  RL_BUILD_MODULE_NAME := jdk.unsupported_jlink_runtime
+  RL_CREATE_PLUGIN_MOD_OUTPUT := $(BUILDTOOLS_OUTPUTDIR)/$(RL_BUILD_MODULE_NAME)
+
+  # We need to use FixPath for correct path handling on Windows. Avoid space
+  # in FixPath call in order to avoid needing to use strip.
+  RL_JIMAGE_PATH_ARG := $(call FixPath,$(JDK_LINK_OUTPUT_DIR)/lib/modules)
+  RL_MOD_PATH_ARG := $(call FixPath,$(IMAGES_OUTPUTDIR)/jmods)
```

Comment:
I'd much rather prefer to use strip and have proper space after commas. This is the approach taken elsewhere in the build system.

```suggestion
  RL_JIMAGE_PATH_ARG := $(strip $(call FixPath, $(JDK_LINK_OUTPUT_DIR)/lib/modules))
  RL_MOD_PATH_ARG := $(strip $(call FixPath, $(IMAGES_OUTPUTDIR)/jmods))
```

Reply:
Thanks! This will also likely change. I'm thinking of just generating the diff file and putting it into `lib/` of the JDK image. That avoids needing to call this build-time only jlink plugin and this `FixPath` magic.

Reply:
I see. I'm sorry if I did not fully understand how the ongoing discussions affected build source changes. 

Maybe I should just put the build part review of this PR on the backburner, and then you can ping me when you think you have a solution that will stick, and I can check how it holds up from a build perspective?

Reply:
I hope the latest version is where we're headed now. I'll ping you for a review once the dust settles. Thanks!